### PR TITLE
feat: add support for receive operation

### PR DESCRIPTION
**Pull request by thulieblack**
1. Update Topic.js Function with subscribe code
2. Extend the AsyncAPI.yml by adding an additional channel
3. Channel comment/views get total views count

solve #4

### CODE REVIEW

**Comment by thulieblack** [comment]
Okay, honestly, I'm not good with this debugging stuff. I tried my best to make half of the code readable and correct.

Challenges:

1. Find a way to get comment/views count as it keeps saying 0
2. I only made sure that we subscribe to comment/views not sure if this is correct
3. Clean-up code after review.
4. Sorry @derberg for the workload 😁😁 I tried

**Comment by thulieblack** [comment]
@derberg the genereted client file


import paho.mqtt.client as mqtt

mqttBroker = "test.mosquitto.org"

class CommentsServiceClient:

  def __init__(self):
              self.client = mqtt.Client()
              self.view_count = 0
              self.subscribe("comment/views")
              self.client.connect(mqttBroker)

  def sendCommentLiked(self, id):
          topic = "comment/liked"
          self.client.publish(topic, id)
  def sendCommentUnliked(self, id):
          topic = "comment/unliked"
          self.client.publish(topic, id)

  def on_message(self, client, userdata, message):
              topic = message.topic
              payload = message.payload.decode("utf-8")
              if topic == "comment/views":
                self.view_count += 1

  def subscribe(self, topic):
              self.client.subscribe(topic)
              self.client.on_message = self.on_message

**Comment by thulieblack** [comment]
okay, so far, the code works without being hardcoded

The next and Final phase is to figure out how the client can stop generating both channels for subscribe and publish functions.

As it stands, this how the current code generates/produces the client file:

```py
import paho.mqtt.client as mqtt

mqttBroker = "test.mosquitto.org"

class CommentsServiceClient:

  def __init__(self):
              self.client = mqtt.Client()
              self.client.connect(mqttBroker)

  def sendCommentViews(self, callback):
          topic = "comment/views"
          self.client.subscribe(topic)
          self.client.message_callback_add(topic, callback)
  def receiveCommentViews(self, id):
          topic = "comment/views"
          self.client.publish(topic, id)

  def sendCommentLiked(self, callback):
          topic = "comment/liked"
          self.client.subscribe(topic)
          self.client.message_callback_add(topic, callback)
  def sendCommentUnliked(self, callback):
          topic = "comment/unliked"
          self.client.subscribe(topic)
          self.client.message_callback_add(topic, callback)
  def receiveCommentLiked(self, id):
          topic = "comment/liked"
          self.client.publish(topic, id)
  def receiveCommentUnliked(self, id):
          topic = "comment/unliked
          self.client.publish(topic, id)

```

File: README.md [hunk 1]
```diff
@@ -39,11 +37,17 @@ Generate code:
 asyncapi generate fromTemplate https://raw.githubusercontent.com/derberg/python-mqtt-client-template/main/test/fixtures/asyncapi.yml https://github.com/derberg/python-mqtt-client-template --output myclient --force-write --param server=dev
 ```
 
-You can also clone this template locally and run `npm test` to check it in action based on examples. Just remember to use some client to verify if tests are really sending messages to the broker:
+You can also clone this template locally and run `npm test` to check it in action based on examples. Just remember to use some client to verify if tests are really sending or receiving messages to the broker.
+
+For sending:
```
Comment (derberg) [inline README.md:42]:
```suggestion
Subscribe to broker to see if test is sending messages properly:
```

File: README.md [hunk 2]
```diff
@@ -39,11 +37,17 @@ Generate code:
...
 
 ```bash
 docker run hivemq/mqtt-cli sub -t comment/liked -h test.mosquitto.org
 ```
+For receiving:
```
Comment (derberg) [inline README.md:47]: Publish to broker to see if test is consuming messages properly

File: package.json [hunk 3]
```diff
@@ -2,18 +2,21 @@
     "name": "python-mqtt-client-template",
     "version": "0.0.1",
     "generator": {
-        "apiVersion": "v1",
+        "apiVersion": "v2",
```
Comment (derberg) [inline package.json:5]:
```suggestion
        "apiVersion": "v3",
```

File: package.json [hunk 4]
```diff
@@ -2,18 +2,21 @@
...
             "server": {
-              "description": "The server you want to use in the code.",
-              "required": true
+                "description": "The server you want to use in the code.",
+                "required": true
             }
         }
     },
     "dependencies": {
+        "@asyncapi/converter": "^1.4.7",
```
Comment (derberg) [inline package.json:19]: remove please

and run `npm i` again

File: test/fixtures/asyncapi.yml [hunk 5]
```diff
@@ -1,38 +1,69 @@
...
+    messages:
+      commentViews:
+        description: >-
+          Message that is being recived with the total number of views in a
+          comment.
+        payload:
+          $ref: '#/components/schemas/commentCount'
+    description: Gets the total number of comment views.
+operations:
+  commentLiked:
```
Comment (derberg) [inline test/fixtures/asyncapi.yml:38]: think about naming operations from a perspective of the name of generated method

File: template/index.js [hunk 6]
```diff
@@ -25,11 +25,15 @@ export default function({ asyncapi, params }) {
       </Text>
 
       <Text indent={2}>
-        <TopicFunction
-          channels={asyncapi.channels().filterByReceive()}
-        />
+        <GenerateSendFunctions operations = {asyncapi.operations()} />
```
Comment (derberg) [inline template/index.js:28]:
```suggestion
        <GenerateSendFunctions operations = {asyncapi.operations().filterBySend()} />
```

I think so :D

File: components/OperationFunction.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
+//Generates and returns a string containing send functions based on provided operations in the yaml file.
+
+export function GenerateSendFunctions({ operations }) {
+  const sendOperations = Array.from(operations).filter(op => op.isSend());
+  const sendDetails = getFunctionDetails(sendOperations, "send");
```
Comment (derberg) [inline components/OperationFunction.js:5]:
```suggestion
  const sendDetails = getFunctionDetails(sendOperations);
```

File: components/OperationFunction.js [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+  });
+
+  return functions;
+}
+
+//Generates and returns a string containing receive functions based on provided operations in the yaml file.
+
+export function GenerateReceiveFunctions({ operations }) {
+  const receiveOperations = Array.from(operations).filter(op => op.isReceive());
+  const receiveDetails = getFunctionDetails(receiveOperations, "receive");
```
Comment (derberg) [inline components/OperationFunction.js:22]:
```suggestion
  const receiveDetails = getFunctionDetails(receiveOperations);
```

File: template/index.js (hunk 6 above)
Comment (thulieblack) [reply template/index.js:28]: I'm leaving this as it is.

**Pull request merged**